src/rust/ide/view/graph-editor/src/component/node/vcs.rs

```diff
@@ -0,0 +1,193 @@
+//! Functionality related to visualising the version control system status of a node.
+
+use crate::prelude::*;
+use ensogl::display::shape::*;
+use crate::component::node as node;
+use ensogl::gui::component::ShapeView;
+use ensogl::display;
+use ensogl::application::Application;
+use enso_frp as frp;
+use ensogl::data::color;
+
+
+
+// ==============
+// === Status ===
+// ==============
+
+/// The version control system status of a node.
+#[derive(Debug,Copy,Clone)]
+pub enum Status {
+    /// The node has not been modified.
+    Unchanged,
+    /// The node was added.
+    Added,
+    /// The node was edited.
+    Edited,
+}
+
+impl Status {
+    fn get_highlight_color_from_style(self, style:&StyleWatch) -> color::Lcha {
+        match self {
+            Status::Unchanged => style.get_color(ensogl_theme::graph_editor::node::vcs::unchanged),
+            Status::Added     => style.get_color(ensogl_theme::graph_editor::node::vcs::added),
+            Status::Edited    => style.get_color(ensogl_theme::graph_editor::node::vcs::edited),
+        }
+    }
+}
+
+impl Default for Status {
+    fn default() -> Self {
+        Status::Unchanged
+    }
+}
+
+
+
+// =======================
+// === Indicator Shape ===
+// =======================
+
+/// Shape used in the status indicator.
```

Comment:
add a note how it looks like - that it is a border around the node with a specific color